### ISSUE TITLE
"Invalid pointer" error after BIOS upgrade

### ISSUE THREAD

**Issue by Patas007** [ethereum-mining/ethminer#1022]
Hello all
on MSI A320M PRO VD/S mobo the BIOS got updated to latest version and all seems to be working fine (running OS Linux Xubuntu) except ethminer. BIOS downgrade does not seem to be possible. I searched and did not find solution for this.

So I tried to use latest "dev" version of ethminer but the error is same.

```
*** Error in '.... /ethminer': free (): invalid pointer: 0x0000...... ***
===== Backtrace: =====
<many lines>
 Aborted (core dumped)
```

MB: MSI A320M PRO VD/S
CPU(APU): A6-9500E
GPU: Sapphire RX570 4GB
RAM: 4GB
I was running ethminer versions 12 and later 13 for couple of month without any troubles until the BIOS got updated.

Any suggestion what to check or how to get rid of this error please ?
The real console output is attached.

Thank you
PAT

[ethminer_invalid_pointer.txt](https://github.com/ethereum-mining/ethminer/files/1934789/ethminer_invalid_pointer.txt)

**Comment by AndreaLanfranchi** [comment]
It happened to me too.
After bios upgrade most of the settings are set back to their default values.
Ensure "Above 4G decoding" is enabled.

**Comment by AndreaLanfranchi** [comment]
No feed back.
Closing.

**Comment by Patas007** [comment]
I was running the ethminer on some older hardware and waited for MSI to provide newer BIOS. Today I was able to upgrade BIOS on mobo but the behaviour is same. Of course the "Above 4G decoding" I've re-enabled in the BIOS after each of the upgrade because they were set to the default values.
Is there a way how to re-open this case? I do not want to open new one (duplicate).
What output do you want me to provide or is there any way how can I help to troubleshoot this issue?
Thank you for your feedback.

**Comment by AndreaLanfranchi** [comment]
I am sorry but it seems it's not an ethminer problem. Apparently is related to your MOBO

**Comment by Patas007** [comment]
Yes. It seems MSI did something within the BIOS that the ethminer stopped working after the upgrade. Previously it worked like a charm. I will try to search for solution how to downgrade the BIOS because officially it's not supported ... sh.t ... :) Anyway now running some other (older) mobo with same HDD and GPU and ethminer works like a charm on it. Thank you for your time. I will not bother you with this "issue" more.

**Comment by Patas007** [comment]
SOLVED. For those who will upgrade BIOS on MSI AM4 mobos and ethminer stops working (it core dumps) follow procedure written in MSI forum here [https://forum-en.msi.com/index.php?topic=302638.0](url)
In my case of mobo A320M PRO-VD/S I had to downgrade to version 7A36v23 (released 31.7.2017). With newer BIOSes ethminer was not working for me.
